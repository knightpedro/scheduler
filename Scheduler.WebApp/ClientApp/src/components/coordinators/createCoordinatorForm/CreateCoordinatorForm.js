import React from "react";
import { BaseForm, FormGroup } from "../../common/forms";
import * as yup from "yup";

const validationSchema = yup.object().shape({
    name: yup
        .string()
        .label("Name")
        .max(50)
        .required(),
    email: yup
        .string()
        .label("Email")
        .email("Invalid email")
        .required(),
});

const initialValues = {
    name: "",
    email: "",
};

const CreateCoordinatorForm = ({ handleSubmit, handleCancel }) => {
    return (
        <BaseForm
            initialValues={initialValues}
            validationSchema={validationSchema}
            onSubmit={handleSubmit}
            onCancel={handleCancel}
        >
            <FormGroup
                name="name"
                className="form-control"
                placeholder="Full Name"
            />
            <FormGroup
                name="email"
                type="email"
                className="form-control"
                placeholder="Email"
            />
        </BaseForm>
    );
};

export default CreateCoordinatorForm;
