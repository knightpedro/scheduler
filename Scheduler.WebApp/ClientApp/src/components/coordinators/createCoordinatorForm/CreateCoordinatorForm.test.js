import React from "react";
import { shallow } from "enzyme";
import CreateCoordinatorForm from "./CreateCoordinatorForm";

describe("CreateCoordinatorForm", () => {
    it("renders correctly", () => {
        const wrapper = shallow(<CreateCoordinatorForm />);
        expect(wrapper).toMatchSnapshot();
    });
});
