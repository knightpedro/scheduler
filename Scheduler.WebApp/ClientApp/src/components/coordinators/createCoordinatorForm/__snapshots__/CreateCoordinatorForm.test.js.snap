// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCoordinatorForm renders correctly 1`] = `
<BaseForm
  initialValues={
    Object {
      "email": "",
      "name": "",
    }
  }
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": undefined,
      "_nodes": Array [
        "email",
        "name",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "email": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "email": false,
            "required": true,
          },
          "_label": "Email",
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "name": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "max": true,
            "required": true,
          },
          "_label": "Name",
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
>
  <FormGroup
    className="form-control"
    name="name"
    placeholder="Full Name"
  />
  <FormGroup
    className="form-control"
    name="email"
    placeholder="Email"
    type="email"
  />
</BaseForm>
`;
